src/components/Filter.js

```diff
@@ -9,14 +12,21 @@ Filter.propTypes = {
 export default function Filter({ setSearchResult }) {
   return (
     <FilterStyled>
-      <InputStyled
-        type="search"
-        onChange={(event) => handleChange(event)}
-        placeholder="Suche nach Name oder PLZ"
+      <Autocomplete
+        options={filterData.map((option) => option.plz)}
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            className="textfield"
+            InputProps={{ ...params.InputProps, type: 'text' }}
+            placeholder="Suche nach PLZ"
+            onSelect={(event) => handleSelect(event)}
```

Comment:
vielleicht kannst du das auch in eine Zeile schreiben (ohne Funktion unten) `{(e) => setSearchResult(e.target.value)}` ?